vendor/bat-native-ads/src/bat/ads/internal/purchase_intent/purchase_intent_classifier_unittest.cc

```diff
@@ -0,0 +1,224 @@
+      no_segments, 0},
+  {"https://creators.brave.com/",
+      no_segments, 0},
+};
+
+}  // namespace
+
+namespace ads {
+
+class AdsPurchaseIntentClassifierTest : public ::testing::Test {
+ protected:
+  std::unique_ptr<MockAdsClient> mock_ads_client_;
+  std::unique_ptr<AdsImpl> ads_;
+
+  AdsPurchaseIntentClassifierTest() :
+      mock_ads_client_(std::make_unique<MockAdsClient>()),
+      ads_(std::make_unique<AdsImpl>(mock_ads_client_.get())) {
+    // You can do set-up work for each test here
+  }
+
+  ~AdsPurchaseIntentClassifierTest() override {
+    // You can do clean-up work that doesn't throw exceptions here
+  }
+
+  // If the constructor and destructor are not enough for setting up and
+  // cleaning up each test, you can use the following methods
+
+  void SetUp() override {
+    // Code here will be called immediately after the constructor (right before
+    // each test)
+    purchase_intent_classifier_ = std::make_unique<PurchaseIntentClassifier>(
+        kPurchaseIntentSignalLevel, kPurchaseIntentClassificationThreshold,
+        kPurchaseIntentSignalDecayTimeWindow);
+
+    // TODO(Moritz Haller): Dynamically generate histories
+    auto now = Time::NowInSeconds();
+    auto days = base::Time::kSecondsPerHour * base::Time::kHoursPerDay;
+
+    auto p1 = PurchaseIntentSignalHistory();
+    p1.timestamp_in_seconds = now - (6 * days);
+    p1.weight = 9;
+    auto p2 = PurchaseIntentSignalHistory();
+    p2.timestamp_in_seconds = now - (5 * days);
+    p2.weight = 9;
+    auto p3 = PurchaseIntentSignalHistory();
+    p3.timestamp_in_seconds = now - (4 * days);
+    p3.weight = 9;
+    auto p4 = PurchaseIntentSignalHistory();
+    p4.timestamp_in_seconds = now - (3 * days);
+    p4.weight = 1;
+    auto p5 = PurchaseIntentSignalHistory();
+    p5.timestamp_in_seconds = now - (2 * days);
+    p5.weight = 1;
+    auto p6 = PurchaseIntentSignalHistory();
+    p6.timestamp_in_seconds = now - (1 * days);
+    p6.weight = 2;
+    auto p7 = PurchaseIntentSignalHistory();
+    p7.timestamp_in_seconds = now - (1 * days);
+    p7.weight = 1;
+    auto p8 = PurchaseIntentSignalHistory();
+    p8.timestamp_in_seconds = now - (1 * days);
+    p8.weight = 1;
+    auto p9 = PurchaseIntentSignalHistory();
+    p9.timestamp_in_seconds = now - (1 * days);
+    p9.weight = 1;
+    auto p10 = PurchaseIntentSignalHistory();
+    p10.timestamp_in_seconds = now - (1 * days);
+    p10.weight = 1;
+    auto p11 = PurchaseIntentSignalHistory();
+    p11.timestamp_in_seconds = now - (1 * days);
+    p11.weight = 1;
+
+    histories_ = {
+      {"cat_5", {p1, p4, p9, p10}},
+      {"cat_2", {p2, p5, p11}},
+      {"cat_1", {p3, p6}},
+      {"cat_4", {p7}},
+      {"cat_3", {p8}}
+    };
+
+    histories_short_ = {
```

Comment:
can this not be declared `const` below under `// Objects declared here can be used by all tests in the test case`